internal/plan/plan.go

```diff
@@ -215,6 +211,27 @@ func (s *Service) Equal(other *Service) bool {
 	return reflect.DeepEqual(s, other)
 }
 
+// LogsTo returns true if the logs from s should be forwarded to target t.
+// This happens if:
+//   - t.Selection is "opt-out" or empty, and s.LogTargets is empty; or
```

Comment:
Good point, and I did discuss this with Ben, but we opted to defer the discussion at the time in favour of more pressing things. I did a [quick test](https://go.dev/play/p/CR6teO5VuRG) and it seems that an unspecified `log-targets` field unmarshals to `[]string(nil)`, while `log-targets: []` unmarshals to `[]string{}`. So it would be possible to make the distinction - in the LogsTo function for example, we would check if `s.LogTargets == nil` instead of checking if `len(s.LogTargets) == 0`. We would just need to be careful about maintaining this distinction, especially when merging - I can see that when `append`ing, it would be easy for a nil slice to turn into an empty slice or vice versa. :)